fhir-avro/src/main/avro/ValueSet.avsc

```diff
@@ -1,60 +1,71 @@
 {
-    "name": "ValueSet",
-    "type": "record",
-    "namespace": "io.carrera.fhir.avro.models",
-    "fields": [
-      {
-        "name": "resourceType",
-        "type": "string"
-      },
-      {
-        "name": "id",
-        "type": "string"
-      },
-      {
-        "name": "name",
-        "type": "string"
-      },
-      {
-        "name": "title",
-        "type": "string"
-      },
-      {
-        "name": "description",
-        "type": "string"
-      },
-      {
-        "name": "status",
-        "type": "string"
-      },
-      {
-        "name": "experimental",
-        "type": "boolean"
-      },
-      {
-        "name": "date",
-        "type": "int",
-        "logicalType": "date"
-      },
-      {
-        "name": "publisher",
-        "type": "string"
-      },
-      {
+  "name": "ValueSet",
+  "type": "record",
+  "namespace": "io.carrera.fhir.avro.models",
+  "fields": [
+    {
+      "name": "resourceType",
+      "type": "string"
+    },
+    {
+      "name": "id",
+      "type": [
+        "null",
+        "string"
+      ],
+      "default": null
+    },
+    {
+      "name": "name",
+      "type": ["null", "string"],
+      "default": null
+    },
+    {
+      "name": "title",
+      "type": ["null", "string"],
+      "default": null
+    },
+    {
+      "name": "description",
+      "type": ["null", "string"],
+      "default": null
+    },
+    {
+      "name": "status",
+      "type": "string"
+    },
+    {
+      "name": "experimental",
+      "type": ["null", "boolean"],
+      "default": null
+    },
+    {
+      "name": "date",
+      "type": ["null", "int"],
+      "logicalType": "date",
+      "default": null
+    },
+    {
+      "name": "publisher",
+      "type": ["null", "string"],
+      "default": null
+    },
+    {
+      "name": "compose",
+      "type": ["null", {
         "name": "compose",
-        "type": {
-          "name": "compose",
-          "type": "record",
-          "fields": [
-            {
-              "name": "include",
-              "type": {
-                "type": "array",
-                "items": ["SimpleCode", "ValueSetPointer"]
-              }
+        "type": "record",
+        "fields": [
+          {
+            "name": "include",
+            "type": {
+              "type": "array",
+              "items": ["SimpleCode", "ValueSetPointer"]
```

Comment:
This isn't conformant to the FHIR spec, but I'm not sure I want to touch it since I'm unsure if it's being used like this.

Reply:
We're working on a transpiler so we can generate these files.
At that point, it will have to conform to the spec.
I think we should merge this as is, fix any fhir->avro mapping issues in the execution engine, then circle back to make it conform to the spec.